modules/tensor_mechanics/include/materials/CappedMohrCoulombCosseratStressUpdate.h

```diff
@@ -0,0 +1,71 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef CAPPEDMOHRCOULOMBCOSSERATSTRESSUPDATE_H
+#define CAPPEDMOHRCOULOMBCOSSERATSTRESSUPDATE_H
+
+#include "CappedMohrCoulombStressUpdate.h"
+
+class CappedMohrCoulombCosseratStressUpdate;
+
+template <>
+InputParameters validParams<CappedMohrCoulombCosseratStressUpdate>();
+
+/**
+ * CappedMohrCoulombCosseratStressUpdate implements rate-independent nonassociative
+ * Mohr-CoulombC plus tensile plus compressive plasticity with hardening/softening
+ * in the Cosserat setting.  The Mohr-Coulomb plasticity is based on the symmetric
+ * part of the stress tensor only, and uses an isotropic elasticity tensor.
+ */
+class CappedMohrCoulombCosseratStressUpdate : public CappedMohrCoulombStressUpdate
+{
+public:
+  CappedMohrCoulombCosseratStressUpdate(const InputParameters & parameters);
+
+  bool requiresIsotropicTensor() override { return false; }
```

Comment:
I've updated the doco here.  The user can utilise an anisotropic elasticity tensor, and usually does since Cosserat is often anisotropic, BUT, only the isotropic parts of it are used in the flow rule.  So ```return false``` is correct.